File: 414-ThirdMaximumNumber/414-ThirdMaximumNumber.py
# Last updated: 7/3/2026, 10:34:08 AM
1class Solution(object):
2    def thirdMax(self, nums):
3        s=set(nums)
4        if(len(s)<3):
5            return max(s)
6        s.remove(max(s))
7        s.remove(max(s))
8        return max(s)
9        

File: 414-ThirdMaximumNumber/414-ThirdMaximumNumber.java
// Last updated: 7/3/2026, 11:23:45 AM
1class Solution {
2    public boolean isHappy(int n) {
3        HashSet<Integer>set=new HashSet<>();
4        while(n!=1 && !set.contains(n)){
5            set.add(n);
6            int s=0;
7            while(n>0){
8                int d=n%10;
9                s+=d*d;
10                n/=10;
11            }
12            n=s;
13        }
14        return n==1;
15    }
16}
17    